Client/Utils/HistoryDeletionUtility.swift

```diff
@@ -13,29 +26,129 @@ class HistoryDeletionUtility {
+                profile.places
+                    .deleteVisitsFor(url: currentSite)
+                    .uponQueue(.global(qos: .userInitiated)) { result in
+                        guard let lastSite = sites.last,
+                              lastSite == currentSite
+                        else { return }
+
+                        continuation.resume(returning: result.isSuccess)
+                    }
+            }
+        }
+    }
+
+    // MARK: - Date based deletion functions
+    private func deleteWKWebsiteDataSince(
+        _ dateOption: HistoryDeletionUtilityDateOptions,
+        for types: Set<String>
+    ) async {
+        guard let date = dateFor(dateOption, requiringAllTimeAsPresent: false) else { return }
+
+        await MainActor.run {
+            WKWebsiteDataStore.default().removeData(
+                ofTypes: types,
+                modifiedSince: date,
+                completionHandler: { }
+            )
+        }
+    }
 
-                completion?(result.isSuccess)
+    private func deleteProfileHistorySince(_ dateOption: HistoryDeletionUtilityDateOptions) async -> Bool? {
+        switch dateOption {
+        case .allTime:
+            return await withCheckedContinuation { continuation in
+                profile.history
+                    .clearHistory()
+                    .uponQueue(.global(qos: .userInteractive)) { result in
+                        continuation.resume(returning: result.isSuccess)
+                    }
             }
+
+        default:
+            guard let date = dateFor(dateOption) else { return nil }
+
+            return await withCheckedContinuation { continuation in
+                profile.history
+                    .removeHistoryFromDate(date)
+                    .uponQueue(.global(qos: .userInteractive)) { result in
+                        continuation.resume(returning: result.isSuccess)
+                    }
+            }
+        }
+    }
+
+    private func deleteProfileMetadataSince(_ dateOption: HistoryDeletionUtilityDateOptions) async -> Bool? {
+        guard let date = dateFor(dateOption) else { return nil }
+        let dateInMilliseconds = date.toMillisecondsSince1970()
+
+        return await withCheckedContinuation { continuation in
+            profile.places
+                .deleteHistoryMetadata(since: dateInMilliseconds)
+                .uponQueue(.global(qos: .userInteractive)) { result in
+                    continuation.resume(returning: result.isSuccess)
+                }
+        }
+    }
+
+    private func clearRecentlyClosedTabs(using dateOption: HistoryDeletionUtilityDateOptions) {
+        switch dateOption {
+        case .allTime:
+            profile.recentlyClosedTabs.clearTabs()
+        default:
+            guard let date = dateFor(dateOption) else { return }
+
+            profile.recentlyClosedTabs.removeTabsFromDate(date)
+        }
+
+        NotificationCenter.default.post(name: .TopSitesUpdated, object: self)
```

Comment:
This notification was previsouly sent on the history deletion of .today or .yesterday options since this case wasn't covered by the .PrivateDataClearedHistory notification top update topsites. I'm not sure this has now the same behavior (I can check that if you want)

Reply:
I'm not sure what that behaviour is/is desired to be. I can check if you tell me the steps, or you can check if it's faster. I merely included what was there, essentially, though there is some difference with the current consolidation.

Reply:
Tag me once the closure changes are done and I'll have a look

Reply:
This also gets called when we choose the .everything option, so I would suggest moving this to where I put my other comment.